central/policy/service/service_impl_postgres_test.go

```diff
@@ -0,0 +1,147 @@
+	categoryPostgres "github.com/stackrox/rox/central/policycategory/store/postgres"
+	edgeDataStore "github.com/stackrox/rox/central/policycategoryedge/datastore"
+	edgeSearch "github.com/stackrox/rox/central/policycategoryedge/search"
+	edgePostgres "github.com/stackrox/rox/central/policycategoryedge/store/postgres"
+	connectionMocks "github.com/stackrox/rox/central/sensor/service/connection/mocks"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	mitreDataStore "github.com/stackrox/rox/pkg/mitre/datastore"
+	"github.com/stackrox/rox/pkg/postgres"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	searchPkg "github.com/stackrox/rox/pkg/search"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/mock/gomock"
+)
+
+func TestPolicyServiceWithPostgres(t *testing.T) {
+	suite.Run(t, new(PolicyServicePostgresSuite))
+}
+
+type PolicyServicePostgresSuite struct {
+	suite.Suite
+	mockCtrl          *gomock.Controller
+	ctx               context.Context
+	db                postgres.DB
+	policies          policyDatastore.DataStore
+	categories        policyCategoryDatastore.DataStore
+	clusters          clusterDatastore.DataStore
+	mitreVectorStore  mitreDataStore.AttackReadOnlyDataStore
+	lifecycleManager  *lifecycleMocks.MockManager
+	connectionManager *connectionMocks.MockManager
+	tested            Service
+}
+
+func (s *PolicyServicePostgresSuite) SetupSuite() {
+
+	s.ctx = sac.WithAllAccess(context.Background())
+	s.db = pgtest.ForT(s.T())
+
+	pStore := policyStore.New(s.db)
+
+	notifierDS := notifierDatastore.GetTestPostgresDataStore(s.T(), s.db)
+
+	categoryStorage := categoryPostgres.New(s.db)
+	categorySearcher := categorySearch.New(categoryStorage)
+
+	edgeStorage := edgePostgres.New(s.db)
+	edgeSearcher := edgeSearch.New(edgeStorage)
+
+	edgeDatastore := edgeDataStore.New(edgeStorage, edgeSearcher)
+
+	s.categories = policyCategoryDatastore.New(categoryStorage, categorySearcher, edgeDatastore)
+
+	s.policies = policyDatastore.New(pStore, search.New(pStore), s.clusters, notifierDS, s.categories)
+
+	var err error
+	s.clusters, err = clusterDatastore.GetTestPostgresDataStore(s.T(), s.db)
+	s.Require().NoError(err)
+
+	s.mitreVectorStore = mitreDataStore.NewMitreAttackStore()
+
+	s.mockCtrl = gomock.NewController(s.T())
+
+	s.lifecycleManager = lifecycleMocks.NewMockManager(s.mockCtrl)
+
+	s.connectionManager = connectionMocks.NewMockManager(s.mockCtrl)
+
+	s.tested = New(s.policies, s.clusters, nil, nil, notifierDS, s.mitreVectorStore, nil, s.lifecycleManager, nil, nil, s.connectionManager)
+}
+
+func (s *PolicyServicePostgresSuite) TearDownSuite() {
+	s.db.Close()
+}
+
+// TestPostPolicy tests posting and then immediately after putting the same policy, as this was an edge case that caused
+// issues with the policy as code config controller.
+func (s *PolicyServicePostgresSuite) TestPostPolicy() {
```

Comment:
nit: TestPutAfterPostPolicy 
Also add the JIRA to the comment and I would suggest rewording as:
as this discovered a bug in the title casing of policy categories causing the policy as code workflow to create a new policy CR with a new policy category to fail